bin/php-build

```
@@ -512,6 +512,35 @@ function definition_package_name() {
     echo "php-$DEFINITION"
 }
 
+# ### autoconf_version
+#
+# Requires an old autoconf version. Valid values are 2.13 and 2.64, which are
+# commonly shipped with most Linux distros.
+function autoconf_version() {
+    local ver="$1"
+    # if the user has set PHP_AUTOCONF, honor it and ignore this command.
+    if [[ -n "${PHP_AUTOCONF}" ]]; then
+        return
+    fi
+    # Debian-like pattern
+    if command -v "autoconf$ver" &> /dev/null; then
+        export PHP_AUTOCONF="autoconf$ver"
+        return
+    fi
+    # RHEL-like pattern
+    if command -v "autoconf-$ver" &> /dev/null; then
+        export PHP_AUTOCONF="autoconf-$ver"
+        return
+    fi
+    # macOS+Homebrew
+    if is_osx && brew --prefix "autoconf@$ver" &> /dev/null; then
+        PHP_AUTOCONF="$(brew --prefix "autoconf@$ver")/bin/autoconf${ver//./}"
+        export PHP_AUTOCONF
+        return
+    fi
+    # no luck, let's hope that system's autoconf is compatible
```

I think we should actually fail loud here. People can stop this line from being hit by setting `PHP_AUTOCONF` explicitly, or by PRing a missing case to this file on GitHub.

---

The default case is `PHP_AUTOCONF=autoconf`. If that is not the correct version, PHP `./configure` itself will fail early with a descriptive error:

```
buildconf: You need autoconf 2.59 or lower to build this version of PHP."
                You are currently trying to use $ac_version"
                Most distros have separate autoconf 2.13 or 2.59 packages."
                On Debian/Ubuntu both autoconf2.13 and autoconf2.59 packages exist."
                Install autoconf2.13 and set the PHP_AUTOCONF env var to "
                autoconf2.13 and try again.
```

So I guess there is no point in duplicating the check and the error message?

---

Right. 👍